### PR TITLE
Fix gametype not shown with local replays

### LINKED ISSUE #732: Game type often not given in local replays screen
image
The games are a mixture of ladder games, FAF beta games and FAF games but very few have the game type present.

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #732

### CODE REVIEW

File: src/main/java/com/faforever/client/replay/ReplayServiceImpl.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ static String guessModByFileName(String fileName) {
       for (Path replayFile : directoryStream) {
         try {
           LocalReplayInfo replayInfo = replayFileReader.parseMetaData(replayFile);
-          FeaturedMod featuredMod = modService.getFeaturedMod(replayInfo.getFeaturedMod()).getNow(FeaturedMod.UNKNOWN);
+          FeaturedMod featuredMod = modService.getFeaturedMod(replayInfo.getFeaturedMod()).get();
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayServiceImpl.java]: getNow is bad as this is connected to a api call here which might or might not finish in time, we could also use a get with timeout, but the api call has a timeout it self

**Review by Geosearchef — DISMISSED** [review]

File: src/main/java/com/faforever/client/replay/ReplayServiceImpl.java (hunk 1 above)
Comment (micheljung) [reply src/main/java/com/faforever/client/replay/ReplayServiceImpl.java]: We shouldn't use getNow() nor get(), but instead use thenAccept(), right?

**Review by micheljung — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/replay/ReplayServiceImpl.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/replay/ReplayServiceImpl.java]: Yeah we should

Comment (1-alex98) [reply src/main/java/com/faforever/client/replay/ReplayServiceImpl.java]: No we can but we don't need to it is all inside a CompletedFutureTask, so I have to call get at some point anyway ;)

File: src/main/java/com/faforever/client/replay/LoadLocalReplaysTask.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public LoadLocalReplaysTask(ReplayService replayService, I18n i18n) {
   @Override
   protected Collection<Replay> call() throws Exception {
     updateTitle(i18n.get("replays.loadingLocalTask.title"));
-    return replayService.getLocalReplays();
+    return replayService.getLocalReplays().get();
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/LoadLocalReplaysTask.java:29]: Now get just shifted further uppwards

Comment (micheljung) [reply src/main/java/com/faforever/client/replay/LoadLocalReplaysTask.java:29]: There has to be a solution for this, let's review it together

Comment (1-alex98) [reply src/main/java/com/faforever/client/replay/LoadLocalReplaysTask.java:29]: ok lets do that

**Pull request closed without merging**